src/collection.rs

```diff
@@ -0,0 +1,127 @@
+/*	libospl - Open Source Collection Library
+	an opensource and multiplateform Collection library management that can be used
+	to store and sort all your Collections.
+	Copyright (C) 2019-2022 Angelo Frangione
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License along
+	with this program; if not, write to the Free Software Foundation, Inc.,
+	51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+use crate::element::ElementDatabase;
+use crate::element::ElementFilesystem;
+use crate::Database;
+use crate::Filesystem;
+use crate::Error;
+
+use chrono::naive::NaiveDateTime;
+
+/// Structure containing a replica of sqlite data
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct Collection
+{
+	pub id:					u32,
+	creation_datetime:		Option<NaiveDateTime>,
+	modification_datetime:	Option<NaiveDateTime>,
+	name:					String,
+	comment:				String,
+}
+
+// Constructors
+impl Collection
+{
+	/// Returns an empty Collection element
+	pub fn new(name: &str, comment: &str) -> Self
+	{
+		Collection
+		{
+			id:						0,
+			creation_datetime:		Some(chrono::offset::Local::now().naive_local()),
+			modification_datetime:	Some(chrono::offset::Local::now().naive_local()),
+			name:					String::from(name),
+			comment:				String::from(comment),
+		}
+	}
+
+	/*pub fn from_folder(&mut self, _db: &Database, collection_path: &str) -> Result <(), Error>
+	{
+		// TODO: Implement.
```

Comment:
I suggest you replace this whole comment section with only a TODO that explains what to implement, because the code might not look like this while we come there.